syntax = "proto3";

package location;

// Define the LocationService with two RPC methods
service LocationService {
  rpc GetLocation (LocationRequest) returns (LocationResponse);
  rpc ListLocations (EmptyRequest) returns (LocationListResponse);
}

// Request for getting a location by ID
message LocationRequest {
  string id = 1;
}

// Empty request (used for listing locations)
message EmptyRequest {}

// Response with a location's data
message LocationResponse {
  string id = 1;
  string name = 2;
  double latitude = 3;
  double longitude = 4;
}

// Response containing a list of locations
message LocationListResponse {
  repeated LocationResponse locations = 1;
}